src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -464,71 +251,32 @@ else if(work.getBaseCaseObj().isUseRegexInstead())
                             }
 
-
-                            rawResponce = "";
                             //build a string version of the headers
-                            rawResponce = httpget.getStatusLine() + "\r\n";
-                            Header[] headers = httpget.getResponseHeaders();
-
-                            StringBuffer buf = new StringBuffer();
-                            for(int a = 0; a < headers.length; a++)
-                            {
-                                buf.append(headers[a].getName() + ": " + headers[a].getValue() + "\r\n");
-                            }
+                            rawResponse = getHeadersAsString(httpMethod);
 
-                            buf.append("\r\n");
-
-                            rawResponce = rawResponce + buf.toString();
-
-                            if(httpget.getResponseContentLength() > 0)
+                            if(httpMethod.getResponseContentLength() > 0)
                             {
 
-                                //get the http body
-                                BufferedReader input = new BufferedReader(new InputStreamReader(httpget.getResponseBodyAsStream()));
-
-                                String line;
-
-                                String tempResponce = "";
-
-                                buf = new StringBuffer();
-                                while((line = input.readLine()) != null)
-                                {
-                                    buf.append("\r\n" + line);
-                                }
-                                tempResponce = buf.toString();
-                                input.close();
-
-
-                                rawResponce = rawResponce + tempResponce;
-
-
-                                Header contentType = httpget.getResponseHeader("Content-Type");
+                                String responseBodyAsString = getResponseAsString(httpMethod);
+                                rawResponse = rawResponse + responseBodyAsString;
 
                                 if(Config.parseHTML)
                                 {
-                                    contentType = httpget.getResponseHeader("Content-Type");
-
-                                    if(contentType != null)
-                                    {
-                                        if(contentType.getValue().startsWith("text"))
-                                        {
-                                            manager.addHTMLToParseQueue(new HTMLparseWorkUnit(tempResponce, work));
-                                        }
-                                    }
+                                    parseHtml(httpMethod, responseBodyAsString);
                                 }
                             }
 
-                            httpget.releaseConnection();
+                            httpMethod.releaseConnection();
                         }
 
 
                         if(work.isDir())
```

Comment:
It can be replaced with:
`notifyItemFound(code, response, rawResponse, null);`
or:
`notifyItemFound(code, response, rawResponse, null, "no fail code");`
to differentiate from other cases.